### PR TITLE
termineter: init at 1.0.4

### PR DESCRIPTION
**Pull request by t4ccer**
## Description of changes

Added [termineter](https://code.google.com/archive/p/termineter/) package and a couple of python dependencies

Tracked in https://github.com/NixOS/nixpkgs/issues/81418

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/te/termineter/package.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ lib
+, fetchFromGitHub
+, python3
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "termineter";
+  version = "1.0.4";
+
+  format = "pyproject";
```
Comment (WolfangAukang) [inline pkgs/by-name/te/termineter/package.nix:10]:
```suggestion
  pyproject = true;
```
#254136

File: pkgs/by-name/te/termineter/package.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  propagatedBuildInputs = with python3.pkgs; [
+    crcelk
+    pluginbase
+    pyasn1
+    pyserial
+    smoke-zephyr
+    tabulate
+    termcolor
+    setuptools
```
Comment (WolfangAukang) [inline pkgs/by-name/te/termineter/package.nix:27]: If the change indicated above affects the building process, try adding setuptools to `propagatedBuildInputs`.

**Review by WolfangAukang — CHANGES_REQUESTED** [review]
Is it possible to build the libraries from source instead of Pypi? `smoke-zephyr` has tests.

**Comment by t4ccer** [comment]
I've changed python sources to github, didn't realize that there is a difference and fetching from pypi won't fetch tests, thanks for catching that

**Comment by WolfangAukang** [comment]
Result of `nixpkgs-review pr 277125` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>python310Packages.crcelk</li>
    <li>python310Packages.crcelk.dist</li>
    <li>python310Packages.smoke-zephyr</li>
    <li>python310Packages.smoke-zephyr.dist</li>
    <li>python311Packages.crcelk</li>
    <li>python311Packages.crcelk.dist</li>
    <li>python311Packages.smoke-zephyr</li>
    <li>python311Packages.smoke-zephyr.dist</li>
    <li>termineter</li>
    <li>termineter.dist</li>
  </ul>
</details>

**Review by WolfangAukang — APPROVED** [review]

**Comment by quantenzitrone** [comment]
Result of `nixpkgs-review pr 277125` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>python310Packages.crcelk</li>
    <li>python310Packages.crcelk.dist</li>
    <li>python310Packages.smoke-zephyr</li>
    <li>python310Packages.smoke-zephyr.dist</li>
    <li>python311Packages.crcelk</li>
    <li>python311Packages.crcelk.dist</li>
    <li>python311Packages.smoke-zephyr</li>
    <li>python311Packages.smoke-zephyr.dist</li>
    <li>termineter</li>
    <li>termineter.dist</li>
  </ul>
</details>

**Review by quantenzitrone — APPROVED** [review]

File: pkgs/by-name/te/termineter/package.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  ];
+
+  meta = with lib; {
+    description = "Smart Meter Security Testing Framework";
+    homepage = "https://code.google.com/p/termineter";
```
Comment (fabaff) [inline pkgs/by-name/te/termineter/package.nix:32]: Google Code was decommissioned years ago :wink: 

```suggestion
    homepage = "https://github.com/rsmusllp/termineter";
```

File: pkgs/development/python-modules/crcelk/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{ buildPythonPackage
+, fetchFromGitHub
+, lib
+}:
+
+buildPythonPackage rec {
+  pname = "crcelk";
+  version = "1.3";
```
Comment (fabaff) [inline pkgs/development/python-modules/crcelk/default.nix:8]:
```suggestion
  version = "1.3";
  format = "setuptools";

  disabled = pythonOlder "3.7";
```

I suggest to limited the support to Python release which are still maintained, especially for new packages as the chance is high that nobody will test or run it against an obsolete Python release.

File: pkgs/development/python-modules/smoke-zephyr/default.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
+{ buildPythonPackage
+, fetchFromGitHub
+, lib
+}:
+
+buildPythonPackage rec {
+  pname = "smoke-zephyr";
+  version = "2.0.1";
```
Comment (fabaff) [inline pkgs/development/python-modules/smoke-zephyr/default.nix:8]:
```suggestion
  version = "2.0.1";
  format = "setuptools";

  disabled = pythonOlder "3.7";
```

**Comment by t4ccer** [comment]
Done in https://github.com/NixOS/nixpkgs/pull/348550

**Pull request closed without merging**